src/model.cpp

```diff
@@ -1072,29 +1074,31 @@ void Model::getEventTimeSensitivity(std::vector<realtype> &stau,
     std::fill(stau.begin(), stau.end(), 0.0);
 
     for (int ip = 0; ip < nplist(); ip++) {
-        fstau(&stau.at(ip), t, x.data(), state_.unscaledParameters.data(),
-              state_.fixedParameters.data(), state_.h.data(), sx.data(ip),
+        fstau(&stau.at(ip), t, computeX_pos(x),
+              state_.unscaledParameters.data(), state_.fixedParameters.data(),
+              state_.h.data(), sx.data(ip),
               plist(ip), ie);
     }
 }
 
 void Model::addStateEventUpdate(AmiVector &x, const int ie, const realtype t,
                                 const AmiVector &xdot,
                                 const AmiVector &xdot_old) {
-
+    
     derived_state_.deltax_.assign(nx_solver, 0.0);
 
     // compute update
-    fdeltax(derived_state_.deltax_.data(), t, x.data(), state_.unscaledParameters.data(),
-            state_.fixedParameters.data(), state_.h.data(), ie, xdot.data(),
-            xdot_old.data());
+    fdeltax(derived_state_.deltax_.data(), t, computeX_pos(x),
+            state_.unscaledParameters.data(), state_.fixedParameters.data(),
+            state_.h.data(), ie, xdot.data(), xdot_old.data());
 
     if (always_check_finite_) {
         app->checkFinite(derived_state_.deltax_, "deltax");
     }
 
     // update
-    amici_daxpy(nx_solver, 1.0, derived_state_.deltax_.data(), 1, x.data(), 1);
+    amici_daxpy(nx_solver, 1.0, derived_state_.deltax_.data(), 1, x.data(),
```

Comment:
For the update, we don't want to use `computeX_pos`?

Reply:
Yes, it think that makes sense!